src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -14,13 +14,44 @@ export interface UseLoadedDataArgs<T> {
    *   onError: (errState) => ReportError(errState.error)
    * })
    */
-  onError?: (state: ErrorState<T, unknown>) => void;
+  onError?: (state: ErrorState<S, E>) => void;
+
+  /**
+   * An optional handler that will be called on successful data load.
+   * Note that LoadedData object typing already allows expression of error status, convenient for consumption by
+   * visual components.  This handler is to accommodate additional side effects within the hook consuming
+   * useLoadedData hook
+   * @param state - the ready state as of when the error happened
```

Comment:
Instead of `the ready state as of when the error happened`, should this be... the state before the last load? The state before the success?

Also, the example usage is for `onError`, not `onSuccess`

Reply:
it will hold the last good ready state data value (success) as of the error happening... could be null if no success has happened yet.

Reply:
fixed in earlier commit